bin/generate.py

```diff
@@ -1017,6 +1052,17 @@ def needed(ext):
             assert ansfile.is_file(), f"Failed to generate ans file: {ansfile}"
             return True
 
+        def generate_empty_interactive_sample_ans():
```

Comment:
why do we need this? the empty `.ans` file is already generated and is used as fallback anyway?

Reply:
1. but (`x.in.statement`, `x.ans`) is still a valid in/ans combination. they don't have to have the exact same extension.
2. that is intended

Reply:
1. Sure, we can also let this function generate `.ans` instead :slightly_smiling_face: 
2. But we cannot manually write an empty `.ans` file in `generators.yaml`, because empty hardcoded file contents are forbidden. And even if this would be allowed, I don't want to manually add empty files for any interactive problem where I only have statement- or download-samples :stuck_out_tongue:

Reply:
Good point, I didn't realize that this function would generate all applicable variants of `.ans*` files :smile: I've rewritten it such that it tries for all variants (in order of priority):
- Whether an answer file already exists, and if so, stop
- Whether a corresponding input file exists, and if so, write an empty answer file and stop

So, for example:
- If `.in`, `.in.statement`, and `.in.download` exist, only an empty `.ans` is generated (if it does not already exist).
- If only `.in.download` exists and `.ans.download` does not, then an empty `.ans.download` is generated.